### PR TITLE
Add YAML Config Files

### PR DESCRIPTION
**Pull request by jonathangreen**
## GitHub Issue

Part of: 
https://github.com/Islandora-CLAW/CLAW/issues/603

Related to:
https://github.com/Islandora-CLAW/Crayfish-Commons/pull/8

## What does this Pull Request do?

Updates all the micro services to use YAML configuration files, instead of the PHP ones we were using before.

## How should this be tested?

- Setup all the microservices using this PR and https://github.com/Islandora-CLAW/Crayfish-Commons/pull/8
- Test the settings in the configuration file and make sure they all work as they did before when the config files were PHP.

# Interested parties
@Islandora-CLAW/committers

### CODE REVIEW

**Comment by whikloj** [comment]
@jonathangreen could you move `Gemini/cfg/config.yaml` to `Gemini/cfg/config.yaml.example` or something so we don't overwrite a person's actual configuration. Saves having to run `git stash save` and `git stash pop` between tests.

**Comment by jonathangreen** [comment]
@whikloj moved.

File: Gemini/src/app.php [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
 use Silex\Application;
 
-$config = require_once(__DIR__ . '/../cfg/cfg.php');
-
 $app = new Application();
-$app->register(new IslandoraServiceProvider($config));
 
-$app['gemini.controller'] = function () use ($app) {
+$app->register(new IslandoraServiceProvider());
+$app->register(new YamlConfigServiceProvider('../cfg/config.yaml'));
```
Comment (jonathangreen) [inline Gemini/src/app.php]: It is fairly hard to give a simple clean error with no logging bootstrapped at this point. Do you have any suggestions for how to accomplish what you would like?

**Comment by jonathangreen** [comment]
Updated the directory where we load the configuration from as suggested.

**Comment by acoburn** [comment]
I don't have any issues with the code here _per se_, but I am concerned about the design. My intention with the original issue was twofold: first, make the configuration YAML-based (rather than based on a PHP array) -- that is clearly solved here; and second, to use the existing Symfony facilities for handling configuration -- that is (from what I can tell) not the case.

There are many advantages of using the Symfony infrastructure for this -- first, you can define a schema for the configuration, so structural errors are clearly caught, and secondly, you can define default values. This means that a user needs only to define the required elements, and the rest can default to sensible values. Why is that important? Imagine a user has Crayfish version 1.2.3 installed with a given configuration, but then updates to version 1.2.4. If that version introduces an optional configuration value, there would need to be no change to the user's config; but as I am reading the code here, it seems that, given this structure, the new version would throw an error. That is _exactly_ what I was trying to avoid when I opened this issue.

**Comment by jonathangreen** [comment]
@acoburn there is more discussion on the other [pull request](https://github.com/Islandora-CLAW/Crayfish-Commons/pull/8). See my response there.

**Comment by whikloj** [comment]
@jonathangreen needs a rebase.

**Pull request merged**